File: dynamic_listings/lua/default/optimization/eager_mode.lua
EagerObject = {}

function EagerObject:new(nums)
    local o = {
        numbers = nums
    }
    setmetatable(o, self)
    self.__index = self
    o["halved_numbers"] = {}
    -- Prepares the halved numbers list
    for i = 1, #nums do
        local num = nums[i]
        table.insert(o["halved_numbers"], num/2)
    end
    return o
end

function EagerObject:getObject(index)
    -- Returns the pre-calculated object at the requested index
    return self.halved_numbers[index];
end

File: dynamic_listings/rust/default/optimization/eager_mode.rs
# #![allow(dead_code)]
struct EagerObject {
    halved_numbers: Vec<u32>,
}

impl EagerObject {
    pub fn new(numbers: &[u32]) -> Self {
        Self {
            // Prepares the halved numbers list
            halved_numbers: numbers.iter().map(|x| x / 2).collect(),
        }
    }

    pub fn object(&self, index: usize) -> u32 {
        // Returns the pre-calculated object at the requested index
        self.halved_numbers[index]
    }
}
